### DISCUSSION TITLE
Week 3 Roundup

### DISCUSSION THREAD

**Discussion by DotNetDublin** [BlazingSaddlesSolutions/TimeKeeper#60] [General]
A productive week for Blazing Solutions and the TimeKeeper team.

Points to note include
- Switched project architecture to adopt a vertical slices approach.
- Added delete time entry function.
- Applied theming to help distinguish between WASM and server side versions of the application.

The above doesn't do justice to the great discussions that revolved around the above and in particular how the switch to vertical slices was implemented. You can check this out [here](url).

This week I think we'll be looking at adding a few simple updates to the time entry screen such as retrieving the clients and categories from the database ensuring that we adhere to the vertical slices approach.

We may then look at mocking the rest of the features that are needed on the time entry screen so these can be discussed ahead of coding.

Finally I'm thinking of adding a roadmap to the project Wiki so each week we can look at a list of features that need to be implemented and agree which should be implemented next.

As always, all feedback welcome!

**Comment by harperjohn** [comment]
@DotNetDublin  Nice "roundup"! And yes - definitely a good and productive week!

I really like the idea of a "roadmap"!! We all have so many ideas/thoughts on features and directions, that it can feel a bit confusing/overwhelming... The roadmap will be a great place to capture these ideas and to put them into some sort of order/priority. It will be nice to have a "summary in one spot" and a "birds-eye-view" of things! Great idea!!! 😃

**Comment by MilesGibson** [comment]
In my last (and very quick) review of the initial application design, I noticed that there is/was no IsBillable flag for a timekeeping entry.  As a consultant (sic) I would like all of my entries to be billable, but that is not always the case, for different reasons.  Might be a good idea to add a boolean flag to reflect this possibility.

**Reply by DotNetDublin** [reply]
Thanks @MilesGibson. Agree a billing makes sense so will add it to the list.
